### DISCUSSION TITLE
The different about algorithm memtest86plus and memtest86

### DISCUSSION THREAD

**Discussion by changpenghui616** [memtest86plus/memtest86plus#335] [Q&A]
Hi 
   I try to use memtest86Plus to test the PC （enter system recovery frequently） ，the result is Pass（2h+）,but use memtest86 （V9.0），it have many error occurred，so the algorithm  whether have some different or shortcoming with the memtest86
thanks

**Comment by x86fr** [comment]
Please post screenshot from both on the same PC.

**Reply by changpenghui616** [reply]
![memtest86(v9 0)](https://github.com/memtest86plus/memtest86plus/assets/106718857/32b90107-3312-46ae-b83b-dac7fe1c81a3)
![memtest86plus](https://github.com/memtest86plus/memtest86plus/assets/106718857/cbd6b70c-39b0-43f2-a739-c15c68dc0f8b)

**Comment by changpenghui616** [comment]
I had upload the screenshot
PS： memtest86 run 29min ，have 385 errors，memtest86plus run 55min，just have 5 errors
by the way，the info （CLK and L1 cache）is different too

**Comment by debrouxl** [comment]
Both programs show the same kind of memory faults, at least :)

Instead of reporting (L1 data cache size + L1 instruction cache size), memtest86+ reports the L1 data cache size, since it's the one that matters for memory accesses performed by tight loops such as the ones we're using.
Also, it does not currently attempt to enable higher processor clock modes, i.e. Turbo boost, or however it's called across processor generations and vendors.

**Comment by changpenghui616** [comment]
right，both programs get memory test fault . But memtest86 can obtain more errors and occurred early，maybe it have some “Special algorithm”  :)

**Comment by x86fr** [comment]
The number of errors detected is not really relevant. 

The delay needed to detect the first error is.  Can you tell how many time needed to detect the first error in both cases?

**Comment by changpenghui616** [comment]
memtest86 first error just need about 5min，and memtest86 plus maybe need 20min or more （the first round pass，second or third round error）

**Reply by x86fr** [reply]
First pass seems to pop an error according to your screenshot.

Can you try to enable ecores to check if errors are detected faster?

**Reply by changpenghui616** [reply]
Im not sure enable ecore whether  is menu：F1/ CPU Selection：/Include E-ocres？
The Default Options is Include E-cores，and i try to modify the option to Exclude E-cores，but it not work（modify finished ，back main menu and retry enter this dir，theoption still is Include-E-cores）

**Reply by x86fr** [reply]
default option is to exclude e-cores : 
https://github.com/memtest86plus/memtest86plus/blob/fc2826726b2281a73647e51418c0130e348eabc2/app/config.c#L90C5-L90C5

The only way to re-enable them at this point is to re-compile the code with `exclude_ecores = false`;

**Reply by changpenghui616** [reply]
When i modify exclude_ecores = false; the fitst error time more later(beyond 3h)
